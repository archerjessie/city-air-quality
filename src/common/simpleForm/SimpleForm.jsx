import React from 'react'
import * as R from 'ramda';
import withState from '../../utils/withState';

const SimpleForm = ({ search, setSearch, onSubmit }) => (
    <form
        className="flex"
        onSubmit={(event) => {
            event.preventDefault();
            onSubmit(search);
        }}>
        <div className="w5 flex items-center">
            <input
                type="text"
                value={search}
                placeholder="Melbourne"
                onChange={R.compose(setSearch, R.prop('value'), R.prop('target'))}
                className="br1 br--top br--left ba b--light-gray ph3 pv2 flex-grow-1 flex-shrink-1"
            />
            <button
                type="submit"
                className="br1 bl-0 br--top br--right b--light-gray white pa2 right-0 flex-grow-0 flex-shrink-0"
            >
                <span role="img" aria-label="search">
                    üïµÔ∏è
              </span>
            </button>
        </div>
    </form>
)
export default R.compose(
    withState('search', 'setSearch', ''),
)(SimpleForm);