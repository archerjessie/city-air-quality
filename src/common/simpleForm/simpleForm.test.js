import React from 'react';
import SimpleForm from './SimpleForm';
import renderer from 'react-test-renderer';

describe("SimpleForm", () => {
    it("renders as expected", () => {
        const tree = renderer
            .create(<SimpleForm search="melbourne" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
})