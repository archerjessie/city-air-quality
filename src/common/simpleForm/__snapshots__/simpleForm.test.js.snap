// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleForm renders as expected 1`] = `
<form
  className="flex"
  onSubmit={[Function]}
>
  <div
    className="w5 flex items-center"
  >
    <input
      className="br1 br--top br--left ba b--light-gray ph3 pv2 flex-grow-1 flex-shrink-1"
      onChange={[Function]}
      placeholder="Melbourne"
      type="text"
      value=""
    />
    <button
      className="br1 bl-0 br--top br--right b--light-gray white pa2 right-0 flex-grow-0 flex-shrink-0"
      type="submit"
    >
      <span
        aria-label="search"
        role="img"
      >
        🕵️
      </span>
    </button>
  </div>
</form>
`;
